app/helpers/application_helper.rb

```diff
@@ -703,6 +722,46 @@ def dark_color_scheme?
     ColorScheme.find_by_id(scheme_id)&.is_dark?
   end
 
+  def forced_light_mode?
+    InterfaceColorSelector.enabled? && cookies[:forced_color_mode] == "light" && !dark_color_scheme?
+  end
+
+  def forced_dark_mode?
+    InterfaceColorSelector.enabled? && cookies[:forced_color_mode] == "dark" && dark_scheme_id != -1
+  end
```

Comment:
If we're changing the HTML response for anonymous users based on a cookie, then we need to add that cookie to the cache-key of the anonymous cache. Otherwise we'll have a (minor) cache-poisoning vulnerability.

However... it would be better if didn't need to do that. Could we do this 100% in JS? The splash-screen already runs a tiny bit of inline JS... so perhaps we could check the cookie in there and set some class on it?

Reply:
I have tried doing this in JS, and from my testing locally switching the media queries in JS comes with UX degradation because the splash screen can "flash" from dark to light and vice versa if the picked color mode doesn't match the OS preference.

I've pushed https://github.com/discourse/discourse/pull/31086/commits/4e058a0845dd1bed1ad164d199ca9c5af5229d71 to include the cookie in the anon cache hey.